databuilder/docs/specs.py

```diff
@@ -93,13 +101,26 @@ def build_paragraph(paragraph_id, test_fn):
     series_line = source_lines[ix + 2]
     series = series_line.strip().removesuffix(",")
 
-    return {
-        "id": paragraph_id,
-        "title": title,
-        "tables": tables,
-        "series": series,
-        "output": output,
-    }
+    # Extract descriptive docstring if any
+    text = inspect.getdoc(test_fn)
+
+    return concatenate_optional_text(
+        {
+            "id": paragraph_id,
+            "title": title,
+            "tables": tables,
+            "series": series,
+            "output": output,
+        },
+        text,
+    )
+
+
+def concatenate_optional_text(dictionary, text):
+    return dict(
+        dictionary,
+        **{"text": text} if text else {},
+    )
```

Comment:
How about...

```suggestion
    return dictionary | {"text": text} if text else dictionary
```

If there is no text, then I think `text` will be an empty string and I think we probably just want to return `dictionary` unchanged, rather than copying it into a new dictionary.

Reply:
If there's no text then `text` will be `None` or possible empty string. 

Yes, I agree it's more efficient not to do the dictionary copy unless necessary. I've never seen the pipe operator used to concat dictionaries like that before and in my testing I get `TypeError: unsupported operand type(s) for |: 'dict' and 'dict'`

Reply:
I think you may be running your tests against Python 3.8 rather than 3.9. I've checked out and amended, and the tests pass. I've also verified that I'm running against 3.9.10 (_.python-version_).

Union operators were added to 3.9 by [PEP 584](https://peps.python.org/pep-0584/).

Reply:
oh, that's **very** nice